perception/autoware_map_based_prediction/src/path_generator.cpp

```diff
@@ -175,17 +175,35 @@ PredictedPath PathGenerator::generatePathForOffLaneVehicle(
 }
 
 PredictedPath PathGenerator::generatePathForOnLaneVehicle(
-  const TrackedObject & object, const PosePath & ref_paths, const double duration,
-  const double lateral_duration, const double speed_limit) const
+  const TrackedObject & object, const PosePath & ref_path, const double duration,
+  const double lateral_duration, const double path_width, const double speed_limit) const
 {
   std::unique_ptr<ScopedTimeTrack> st_ptr;
   if (time_keeper_) st_ptr = std::make_unique<ScopedTimeTrack>(__func__, *time_keeper_);
 
-  if (ref_paths.size() < 2) {
+  if (ref_path.size() < 2) {
+    return generateStraightPath(object, duration);
+  }
+
+  // if the object is moving backward, we generate a straight path
+  if (object.kinematics.twist_with_covariance.twist.linear.x < 0.0) {
     return generateStraightPath(object, duration);
   }
 
-  return generatePolynomialPath(object, ref_paths, duration, lateral_duration, speed_limit);
+  // get object width
+  double object_width = 5.0;  // a large number
+  if (
+    object.shape.type == autoware_perception_msgs::msg::Shape::BOUNDING_BOX ||
+    object.shape.type == autoware_perception_msgs::msg::Shape::CYLINDER) {
```

Comment:
[to be fixed]
CYLINDER object won't have `y` value. Please use `x` for CYLINDER.

```idl
      @verbatim (language="comment", text=
        " x: the length of the object (BOUNDING_BOX) or diameter (CYLINDER)"
        " y: the width of the object (BOUNDING_BOX)"
        " z: the overall height of the object")
```

Reply:
resolved https://github.com/autowarefoundation/autoware.universe/pull/8602/commits/92851eed0e079ec09e1b415f5e1d39d39f9487bb